src/ast/value.rs

```diff
@@ -184,12 +184,31 @@ pub struct EscapeQuotedString<'a> {
 
 impl<'a> fmt::Display for EscapeQuotedString<'a> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        for c in self.string.chars() {
-            if c == self.quote {
-                write!(f, "{q}{q}", q = self.quote)?;
-            } else {
-                write!(f, "{c}")?;
+        let quote = self.quote;
```

Comment:
Can you explain what this change is for?

Reply:
I should've explained this part, thank you!

In short, the change is necessary to correctly re-serialize value which can be either escaped or non-escaped.
Let me explain in order. :)

1. We are now using the same type of `Token` in both the `escaped` and the `no-escape` modes.
This means tokens such as `DoubleQuotedString` can have both escaped and non-escaped strings.

2. On the other hand, `Value` doesn't know which mode of escape is chosen by the user. So `fmt()` for `Value` has to be able to correctly display strings without knowing the strings are escaped or not.

3. The below table explains how `fmt()` should work.
According to it, we can say as follows:
"if the quote symbol in the string is repeated twice, OR, if the quote symbol is after backslash, `fmt()` should display all the chars without any escape. And if the quote symbol is used just between usual chars, `fmt()` should display it twice."

| phrase of original query  | mode | node of AST | serialized |
| ------------- | ------------- | ------------- | ------------- |
| `"A""B""A"`  | no-escape | `DoubleQuotedString(String::from("A\"\"B\"\"A"))`  | `"A""B""A"`  |
| `"A""B""A"`  | default | `DoubleQuotedString(String::from("A\"B\"A"))`  | `"A""B""A"`  |
| `"A\"B\"A"`  | no-escape | `DoubleQuotedString(String::from("A\\\"B\\\"A"))`  | `"A\"B\"A"`  |
| `"A\"B\"A"`  | default | `DoubleQuotedString(String::from("A\"B\"A"))` | `"A""B""A"`  |

This is a bit tricky, but i thought it was the best way for realizing what we discussed...!
would like to hear your frank opinion. :)